### PR TITLE
Add MDGallery

### LINKED ISSUE #192: Gallery
**As a** user,\
**I want** a gallery app like the one by [Flutter](https://flutter.github.io/gallery/)\
**so that** I can view all widgets in all possible configurations.

### PR DESCRIPTION
**Pull request by ctiedt**
Closes #192

### CODE REVIEW

File: packages/Material-Examples.package/MDGallery.class/instance/checkboxes.st [hunk 1]
```diff
@@ -0,0 +1,7 @@
+as yet unclassified
+checkboxes
+
+	^ self wrapper
+		addMorphBack: (self checkboxWithText: 'Option A');
+		addMorphBack: (self checkboxWithText: 'Ich hasse Squeak');
```
Comment (IsabellKraus) [inline packages/Material-Examples.package/MDGallery.class/instance/checkboxes.st:6]: please correct second text ...

File: packages/Material-Examples.package/MDGallery.class/instance/button.st [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	wrapper
+		addMorphBack: MDButton new;
+		addMorphBack: self spacer;
+		addMorphBack: button;
+		addMorphBack: self spacer;
+		addMorphBack: buttonIcon;
+		addMorphBack: self spacer;
+		addMorphBack: buttonDisabled.
+	
+	^ wrapper
```
Comment (JonasWanke) [inline packages/Material-Examples.package/MDGallery.class/instance/button.st:32]:
```suggestion
	^ self wrapper
		addMorphBack: MDButton new;
		addMorphBack: self spacer;
		addMorphBack: (MDButton new
			text: 'Secondary';
			textColor: (self currentTheme textColorOn: self currentTheme secondaryColor);
			backgroundColor: self currentTheme secondaryColor;
			borderRadius: (MDBorderRadius left: 2 right: 8);
			yourself);
		addMorphBack: self spacer;
		addMorphBack: ((MDButton icon: #ic_shopping_cart)
			text: 'Add to cart'
			yourself);
		addMorphBack: self spacer;
		addMorphBack: (MDButton new
			text: 'Disabled';
			disable;
			yourself);
		yourself
```

File: packages/Material-Examples.package/MDGallery.class/instance/buildShowCaseSelection.st [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		item := (MDListItem newOnList: list)
+			text: each key;
+			setProperty: #value toValue: each value;
+			yourself.
+		list addListItem: item].
+	list canDeselect: true.
+
+	^ list
+		
+
```
Comment (JonasWanke) [inline packages/Material-Examples.package/MDGallery.class/instance/buildShowCaseSelection.st:25]:
```suggestion
	^ list
```

File: packages/Material-Examples.package/MDGallery.class/instance/buildContentMorph.st [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+		layoutPolicy: ProportionalLayout new;
+		addMorph: self buildShowCaseSelection
+		fullFrame: (LayoutFrame
+			fractions: (0@0 corner: self class contentPaneWidthFraction@1)
+			offsets: (0@0 corner: self class contentPaneSpacingHorizontal / 2 negated@0));
+		addMorph: self showCaseMorph
+		fullFrame: (LayoutFrame
+			fractions: (self class contentPaneWidthFraction@0 corner: self class contentPaneWidthFraction * 3@1)
+			offsets: (self class contentPaneSpacingHorizontal / 2@0 corner: self class contentPaneSpacingHorizontal / 2 negated@0))
+
```
Comment (JonasWanke) [inline packages/Material-Examples.package/MDGallery.class/instance/buildContentMorph.st:16]:
```suggestion
			offsets: (self class contentPaneSpacingHorizontal / 2@0 corner: self class contentPaneSpacingHorizontal / 2 negated@0))
```

File: packages/Material-Examples.package/MDGallery.class/instance/list.st [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+	list addListItem: (
+			item
+			text: 'ListItem ', each;
+			yourself)].
+		
+	list canDeselect: false.
+
+	^ list
+		
+
```
Comment (JonasWanke) [inline packages/Material-Examples.package/MDGallery.class/instance/list.st:19]:
```suggestion
	^ list
```

File: packages/Material-Examples.package/MDGallery.class/instance/showCaseMorph.st [hunk 6]
```diff
@@ -0,0 +1,7 @@
+as yet unclassified
+showCaseMorph
+
+	^ showCaseMorph
+		ifNil: [showCaseMorph := Morph new
+			color:  Color transparent;
+			layoutPolicy: ProportionalLayout new; yourself]
```
Comment (JonasWanke) [inline packages/Material-Examples.package/MDGallery.class/instance/showCaseMorph.st:7]:
```suggestion
	^ showCaseMorph ifNil: [
		showCaseMorph := Morph new
			color:  Color transparent;
			layoutPolicy: ProportionalLayout new;
			yourself]
```

File: packages/Material-Examples.package/MDGallery.class/instance/textArea.st [hunk 7]
```diff
@@ -0,0 +1,11 @@
+as yet unclassified
+textArea
+
+	^ self wrapper
+		addMorphBack: MDTextArea new;
+		addMorphBack: (self spacerWithSpacing: 32);
+		addMorphBack: (MDTextArea
+						label: 'Name'
+						placeholder: 'What do people call you'
+						helper: 'Lastname not firstname');
```
Comment (JonasWanke) [inline packages/Material-Examples.package/MDGallery.class/instance/textArea.st:10]:
```suggestion
		addMorphBack: (MDTextArea
			label: 'Name'
			placeholder: 'What do people call you'
			helper: 'Lastname not firstname';
			yourself);
```

File: packages/Material-Examples.package/MDGallery.class/class/showCaseList.st [hunk 8]
```diff
@@ -0,0 +1,11 @@
+as yet unclassified
```
Comment (DasSepp) [inline packages/Material-Examples.package/MDGallery.class/class/showCaseList.st:1]: please add category

File: packages/Material-Core.package/MaterialCoreServiceProvider.class/class/initialize.st [hunk 9]
```diff
@@ -0,0 +1,3 @@
+initialization
+initialize 
+	ServiceRegistry current buildProvider: self new
```
Comment (DasSepp) [inline packages/Material-Core.package/MaterialCoreServiceProvider.class/class/initialize.st:3]: please insert a newline here

File: packages/Material-Examples.package/MDGallery.class/instance/dialog.st [hunk 10]
```diff
@@ -0,0 +1,18 @@
+initialization
+dialog
+
+	|dialog|
+
```
Comment (DasSepp) [inline packages/Material-Examples.package/MDGallery.class/instance/dialog.st]: please remove the newline in line 5

File: packages/Material-Examples.package/MDGallery.class/instance/dialog.st [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+dialog
+
+	|dialog|
+	
+	dialog := MDDialog new
+		title: 'Basic Dialog';
+		content: 'This is just a basic dialog.';
+		positiveText: 'OK';
+		yourself.
+
```
Comment (DasSepp) [inline packages/Material-Examples.package/MDGallery.class/instance/dialog.st:10]: please remove the newline in line 11

File: packages/Material-Examples.package/MDGallery.class/instance/menu.st [hunk 12]
```diff
@@ -0,0 +1,7 @@
+as yet unclassified
```
Comment (DasSepp) [inline packages/Material-Examples.package/MDGallery.class/instance/menu.st]: please use a category and change the newlines

File: packages/Material-Examples.package/MDGallery.class/instance/initialize.st [hunk 13]
```diff
@@ -0,0 +1,9 @@
+initialization
+initialize
+
+	self labelString: 'Material Design Gallery'.
+	super initialize.
+
```
Comment (DasSepp) [inline packages/Material-Examples.package/MDGallery.class/instance/initialize.st:6]: please remove the newline

**Pull request merged**